lib/galaxy/util/string_conversion.py

```diff
@@ -0,0 +1,15 @@
+"""
+Safe conversion to unicode
+"""
+
+
+def to_unicode( a_string ):
+    """
+    Convert a string to unicode in utf-8 format; if string is already unicode,
+    does nothing because string's encoding cannot be determined by introspection.
+    """
+    a_string_type = type( a_string )
+    if a_string_type is str:
+        return unicode( a_string, 'utf-8' )
+    elif a_string_type is unicode:
+        return a_string
```

Comment:
> It will not raise an exception, it will just return nothing.

ah yes, sorry, I got this mixed up.

> Please also keep in mind that everything in lib/galaxy/util/ should be Python3-safe, this function is not (as spotted by py34-lint tox in TravisCI).

Thanks, I will check this.